yapapi/engine.py

```diff
@@ -448,26 +454,109 @@ async def _process_invoices(self) -> None:
             ):
                 break
 
+    @staticmethod
+    def _reason_for_termination(
```

Comment:
It's not clear that "reason is `None` == no reason == do not terminate", I thought this is either "known reason" or "unknown reason", but that the termination is already determined.

I'm not saying the implementation is wrong, but a more meaningful name (`check_for_termination_reason` or so) would be nice.

Reply:
will update